Cython/Compiler/Parsing.py

```diff
@@ -3135,7 +3135,8 @@ def p_c_enum_definition(s, pos, ctx):
     s.next()
 
     scoped = False
-    if s.sy in ('class', 'struct'):
+
+    if s.systring in ('class', 'struct'):
```

Comment:
I'd rather have it explicit.
```suggestion
    if s.sy == 'class' or (s.sy == 'IDENT' and s.systring == 'struct'):
```
BTW, I wonder if there's code out there that previously used a `cdef enum struct: …` or `cdef enum struct "some_cname": …`.
Maybe we should be nice and keep that name valid, just in case… So, if there is a `cdef enum struct` with no name, the user probably meant `struct` to be the name. Difficult to say if that's more helpful, more confusing, or more prone to errors, though…

Reply:
Good point. Personally I think we should disallow it. Upon reading `cdef enum struct: ...` one may wonder whether it is:

1. a plain `enum` named `struct`, or
2. an "anonymous `enum struct`"

Of course there is no such thing as (2), but it's not hard to imagine this as being a source of confusion.